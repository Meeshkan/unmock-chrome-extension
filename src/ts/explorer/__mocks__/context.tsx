import * as React from "react";

const actions = {};
export const ExplorerActionsContext = React.createContext({ actions });
