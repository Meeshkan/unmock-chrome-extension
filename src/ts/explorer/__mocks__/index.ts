import { Active, Labeled, Phase } from "../../state";
import { ExplorerState } from "../explorer";

export const mockActions = {
  triggerFetchSuccess: jest.fn(),
  triggerSetActiveUrl: jest.fn(),
  triggerInitializeStore: jest.fn(),
  triggerDownload: jest.fn(),
};

export const mockLabeled: Labeled = {
  "https://docs.readthedocs.io/en/stable/api/v2.html": {
    "/api/v2/project/": {
      GET: {},
      PUT: {},
    },
    "/api/v2/user/": {
      POST: {},
    },
  },
  "https://docs.readthedocs.io/en/stable/api/v1.html": {
    "/api/v1/project/": {
      GET: {},
      PUT: {},
    },
    "/api/v1/user/": {},
  },
};

export const mockActive: Active = {
  activePath: [],
  phase: Phase.ADD_PATH,
};

export const mockExplorerState: ExplorerState = {
  data: {
    active: mockActive,
    labeled: mockLabeled,
  },
  isLoading: false,
};
