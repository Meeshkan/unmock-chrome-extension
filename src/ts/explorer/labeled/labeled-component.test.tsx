import * as React from "react";
import Labeled from "./labeled-component";
import { shallow } from "enzyme";
import { mockActions, mockActive, mockLabeled } from "../__mocks__";

const props = {
  actions: mockActions,
  labeled: mockLabeled,
  active: mockActive,
};

describe("Labeled", () => {
  test("renders correctly", () => {
    const wrapper = shallow(<Labeled {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
