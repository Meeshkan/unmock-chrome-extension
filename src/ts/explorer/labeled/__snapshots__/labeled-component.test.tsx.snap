// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labeled renders correctly 1`] = `
<div>
  <div
    id="labeled"
  >
    <h3>
      Labeled URLs
    </h3>
    <ul>
      <Alert
        closeLabel="Close alert"
        key="https://docs.readthedocs.io/en/stable/api/v2.html"
        show={true}
        transition={[Function]}
        variant="info"
      >
        <div
          onClick={[Function]}
        >
          https://docs.readthedocs.io/en/stable/api/v2.html
        </div>
        <ul>
          <li
            key="/api/v2/project/"
          >
            /api/v2/project/
            <ul>
              <li
                key="GET"
              >
                GET
              </li>
              <li
                key="PUT"
              >
                PUT
              </li>
            </ul>
          </li>
          <li
            key="/api/v2/user/"
          >
            /api/v2/user/
            <ul>
              <li
                key="POST"
              >
                POST
              </li>
            </ul>
          </li>
        </ul>
      </Alert>
      <Alert
        closeLabel="Close alert"
        key="https://docs.readthedocs.io/en/stable/api/v1.html"
        show={true}
        transition={[Function]}
        variant="info"
      >
        <div
          onClick={[Function]}
        >
          https://docs.readthedocs.io/en/stable/api/v1.html
        </div>
        <ul>
          <li
            key="/api/v1/project/"
          >
            /api/v1/project/
            <ul>
              <li
                key="GET"
              >
                GET
              </li>
              <li
                key="PUT"
              >
                PUT
              </li>
            </ul>
          </li>
          <li
            key="/api/v1/user/"
          >
            /api/v1/user/
            <ul />
          </li>
        </ul>
      </Alert>
    </ul>
  </div>
</div>
`;
